### PR TITLE
Fix comparison subquery where prefix comparison operator is used

### PR DESCRIPTION
**Pull request by rayslxu**

### CODE REVIEW

File: lib/ast/expression.ts [hunk 1]
```diff
@@ -1076,7 +1076,10 @@ export class ComparisonSubqueryBooleanExpression extends BooleanExpression {
     }
 
     toSource() : TokenStream {
-        return List.concat(this.lhs.toSource(), this.operator, 'any', '(', this.rhs.toSource(), ')');
+        if (INFIX_COMPARISON_OPERATORS.has(this.operator))
+            return List.concat(this.lhs.toSource(), this.operator, 'any', '(', this.rhs.toSource(), ')');
```
Comment (gcampax) [inline lib/ast/expression.ts:1080]: Do you need to account for operator priority here?

File: test/test_syntax.tt [hunk 2]
```diff
@@ -1921,3 +1921,7 @@ monitor(@com.twitter.home_timeline() filter any(sort(file_size asc of @com.dropb
 monitor(@com.twitter.home_timeline() filter any(@com.twitter.post()));
 
 
+====
+
+// comparison subquery infix comparison operator
+@org.wikidata.city() filter in_array(id, any([shares_border_with] of @org.wikidata.city() filter id =~ "palo alto"));
```
Comment (gcampax) [inline test/test_syntax.tt:1927]: Uh? This test case is not infix?

Comment (rayslxu) [reply test/test_syntax.tt:1927]: Ugh I mean postfix for this whole PR... I will update this once I got back

Comment (gcampax) [reply test/test_syntax.tt:1927]: You mean prefix...

Comment (rayslxu) [reply test/test_syntax.tt:1927]: > You mean prefix...

yes...

**Review by gcampax — APPROVED** [review]
Yep

**Pull request merged**